extend fix splits tool to report splits with mismatched amounts

matt-fidd opened the pull request.

matt-fidd commented on packages/loot-core/src/server/tools/app.ts:
```diff
@@ -54,9 +57,27 @@
+    )
+  ).data;
+
+  const mismatchedSplits = splitTransactions.filter(t => {
+    const subValue = t.subtransactions.reduce((acc, st) => acc + st.amount, 0);
+
+    return subValue !== t.amount;
+  });
```
The amounts are stored as whole numbers, not decimals

matt-fidd commented on packages/desktop-client/src/components/settings/FixSplits.tsx:
```diff
@@ -70,11 +79,7 @@ export function FixSplits() {
-            maxWidth: 500,
-            width: '100%',
-            alignItems: 'center',
+            display: 'block',
```
Fixed in https://github.com/actualbudget/actual/commit/9b12556bd5c42f11eaf31234c9809330391e0fd1 now

UnderKoen reviewed (commented):
Shouldn't we create the missing split for them with an note to notify them?

matt-fidd commented:
> Shouldn't we create the missing split for them with an note to notify them?

I wouldn't want to alter the transactions potentially years back in the data without knowing what the "correct" action is. The end goal here is a modal that pops up and allows the user to correct it there and then, but this is an improvement for the moment.

jfdoming commented on packages/desktop-client/src/components/settings/FixSplits.tsx:
```diff
@@ -15,40 +15,47 @@ import { Setting } from './UI';
+      result.push(
+        `Found ${mismatchedSplits.length} split transaction${mismatchedSplits.length > 1 ? 's' : ''} ` +
+          `with mismatched amounts on the below date${mismatchedSplits.length > 1 ? 's' : ''}. ` +
+          `Please fix ${mismatchedSplits.length > 1 ? 'these' : 'this'} manually:\n` +
```
```suggestion
          `Please review ${mismatchedSplits.length > 1 ? 'them' : 'it'} manually:\n` +
```

jfdoming reviewed (dismissed):
LGTM! If you have time, it would be nice to translate the strings in this file

matt-fidd commented:
> LGTM! If you have time, it would be nice to translate the strings in this file

I've added translations to the dynamic part, and cleaned up the rest of the file into a single <Trans> component. The tenses in the bullet points didn't make much sense in isolation so I've opted to group them for easier understanding. I can split them out if you disagree though.

jfdoming reviewed (dismissed):
Thanks! One note is that for things like tenses, it's usually fine to put them in one string as the translation software splits them into "one" and "many" strings if you pass a count

matt-fidd commented:
> Thanks! One note is that for things like tenses, it's usually fine to put them in one string as the translation software splits them into "one" and "many" strings if you pass a count

Thanks, fixed to use `{{count}}` and only kept the plural as the docs recommend.

jfdoming reviewed (approved):
🙌
